lib/tasks/reporting.rake

```diff
@@ -83,6 +83,22 @@ namespace :reporting do
     send_all_resources(runs, ReportService::RunSender, opts)
   end
 
+  desc "publish anonymous runs to report service"
+  task :publish_anonymous_runs => :environment do
+    runs = Run.where('remote_endpoint is null')
+    # allow caller do specify the limit and offset of the query
+    limit_env_value = ENV["REPORT_PUSH_RUN_LIMIT"]
+    if limit_env_value && limit_env_value.size > 0
```

Comment:
You could use `limit_env_value.present?` instead of checking for the size of the string.

Reply:
I initially copied that from the original rake task but have since deleted it completely as it doesn't play well with the `find_in_batches` methods own limit/offset calculation

Reply:
I was wondering about that. So maybe it should be limited by a date. Or you could add an option to send_all_runs which limits the number of batches.

Reply:
Unfortunately limits where only added to find_in_batches in Rails 5.1:

https://www.bigbinary.com/blog/rails-5-1-adds-support-for-limit-in-batch-processing

Lara is on 3.2.22 right now.

Reply:
I added min/max id options to the anonymous run rake task in this commit (plus I fixed a few typos in the existing code):

https://github.com/concord-consortium/lara/pull/782/commits/16491ed9bbadb3d9a861b37367c77a11a2b03354